README.md

```diff
@@ -136,13 +139,17 @@ allow-licenses:
   - 'GPL-3.0'
   - 'BSD-3-Clause'
   - 'MIT'
+allow-dependencies-licenses:
+  npm: @actions/core, @actions/github
+  pip: requests
 ```
 
 ### Considerations
 
 - Checking for licenses is not supported on Enterprise Server.
 - The action will only accept one of the two `license` parameters; an error will be raised if you provide both.
 - We don't have license information for all of your dependents. If we can't detect the license for a dependency **we will inform you, but the action won't fail**.
+- The `allow-dependencies-licenses` option can only be used with the external configuration file.
```

Comment:
I kept this only to be possible via the `config-file` for two reasons:
- Configuring this inline would make the workflow file get more complicated and I tried to avoid that.
- If I try to pass an object in Actions on that level in the same format I get a syntax error that an `object` is not allows on that level. Having different formats for the same option when configuring it inline and config sounded like a bad idea. 😄 
```
      - name: 'Dependency Review'
        uses: theztefan/dependency-review-action@allow-list-dependencies
        with:
          fail-on-severity: 'critical'
          allow-licenses: MIT, BSD-3-Clause
          comment-summary-in-pr: true
          config-file: thez-org/open-book/dependancy-review/org-wide.yml@main
          allow-dependencies-licenses:                  <--- "Invalid type found: one of string , number , boolean were expected but an object was found"
            npm: "@actions/core, @actions/github"
            pip: "requests
```
Happy to revisit this if you have a suggestion.

Reply:
It's a shame we can't use maps in workflow files.

Something that your example brought to my attention is that we don't have a list of possible ecosystems, and I don't want to be maintaining one every time the Dependency Graph adds a new ecosystem.

I think that using [purl](https://github.com/package-url/purl-spec) is the way to go here, I see a few advantages:
1. It's already being supported by multiple other software supply chain initiatives (including SPDX, which is what we use for licenses).
2. It'd allow us to keep supporting the same configuration both inline and externally.
3. Someone else takes care of defining what the ecosystem keys are.

An inline copy of your example would look like this:

```
allow-dependencies-licenses: "pkg:npm/@actions/core, pkg:npm/@actions/github, pkg:pip/requests"
```

I guess the external one would be:

```
allow-dependencies-licenses:
  - 'pkg:npm/@actions/core'
  - 'pkg:npm/@actions/github'
  - 'pkg:pip/requests'
```

What do you think?

Reply:
Love the idea! I switched it to `purl` format. So, it supports both config file and inline configuration.